src/NuGet.Clients/NuGet.PackageManagement.UI/Options/ConfigPathsWindowViewModel.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Linq;
+using Microsoft.VisualStudio.ComponentModelHost;
+using Microsoft.VisualStudio.Services.Common;
+using NuGet.Common;
+using NuGet.ProjectManagement;
+using NuGet.VisualStudio;
+
+namespace NuGet.PackageManagement.UI.Options
+{
+    public class ConfigPathsWindowViewModel
+    {
+        public ObservableCollection<ConfigPathsViewModel> ConfigPaths { get; private set; }
+
+        public void SetConfigPaths()
+        {
+            IComponentModel componentModelMapping = NuGetUIThreadHelper.JoinableTaskFactory.Run(ServiceLocator.GetComponentModelAsync);
+            var settings = componentModelMapping.GetService<Configuration.ISettings>();
+            IReadOnlyList<string> configPaths = settings.GetConfigFilePaths().ToList();
+            ConfigPaths.AddRange(CreateViewModels(configPaths));
+        }
+
+        private ObservableCollection<ConfigPathsViewModel> CreateViewModels(IReadOnlyList<string> configPaths)
+        {
+            var configPathsCollection = new ObservableCollection<ConfigPathsViewModel>();
```

Comment:
```suggestion
            var configPathsCollection = new ObservableCollection<ConfigPathsViewModel>(configPaths.Count);
```
[doc](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.ireadonlycollection-1.count?view=net-7.0#system-collections-generic-ireadonlycollection-1-count)

Reply:
Hi Kartheek! I looked over the documentation, but I don't understand how to use `configPaths.Count` here. In this line, I'm just creating a new empty `ObservableCollection`. Later on, I create `ConfigPathsViewModel`s from the list of `configPaths`, which I add to the `ObservableCollection`. Do you mean that it would be better to use a for loop based on the count of `configPaths` instead of a foreach loop?